cargo-quickinstall/src/command_ext.rs

```diff
@@ -0,0 +1,111 @@
+use std::{
+    ffi::OsStr,
+    fmt,
+    process::{self, Child, Command, Output},
+};
+
+use crate::{utf8_to_string_lossy, CommandFailed, InstallError};
+
+pub trait CommandExt {
+    fn formattable(&self) -> CommandFormattable<'_>;
+
+    fn output_checked_status(&mut self) -> Result<Output, InstallError>;
+
+    fn spawn_with_cmd(self) -> Result<ChildWithCmd, InstallError>;
+}
+
+impl CommandExt for Command {
+    fn formattable(&self) -> CommandFormattable<'_> {
+        CommandFormattable(self)
+    }
+
+    fn output_checked_status(&mut self) -> Result<Output, InstallError> {
+        self.output()
+            .map_err(InstallError::from)
+            .and_then(|output| check_status(self, output))
```

Comment:
> Could make output_checked_status() create a ChildWithCmd and call wait_with_output_checked_status() on it? That way, you can inline check_status() into it.

I tried but failed because `Command::spawn_with_cmd` calls `Command::spawn`, which by default inherits stdin, stdout and stderr.

While `Command::output_checked_status` calls `Command::output`, which by default read stdout and stderr into a pipe.